Newsletters: add 218 (2022-09-21) (#863)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-09-21-newsletter.md

---
title: 'Bitcoin Optech Newsletter #218'
permalink: /en/newsletters/2022/09/21/
name: 2022-09-21-newsletter
slug: 2022-09-21-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Creating drivechains with APO and a trusted setup:** Jeremy Rubin
  [posted][rubin apodc] to the Bitcoin-Dev mailing list the description
  for how a trusted set up procedure could be combined with the proposed
  [SIGHASH_ANYPREVOUT][topic sighash_anyprevout] opcode to implement
  behavior similar to that proposed by [drivechains][topic sidechains].

>   Drivechains are a type of sidechain where miners are normally in
>   charge of keeping the sidechain funds secure (compared to full nodes
>   which are in charge of securing funds on Bitcoin's mainchain).  Miners

Reviewer: ```suggestion
  Drivechains are a type of sidechain where miners are normally responsible
  for keeping the sidechain funds secure (in contrast to full nodes,
  which secure funds on Bitcoin's mainchain).  Miners
```

>   attempting to steal drivechain funds are required to announce their

Reviewer: I don't understand the concept very well, but maybe:
```suggestion
  attempting to steal drivechain funds unavoidably reveal their
```

  malicious intentions days or weeks in advance, giving users a chance
  to change their full nodes to enforce the rules of the sidechain.
  Drivechains are primarily proposed for inclusion into Bitcoin as a
  soft fork (see BIPs [300][bip300] and [301][bip301]), but a previous
  post to the mailing list (see [Newsletter #190][news190 dc]) described
  how some other flexible proposed additions to Bitcoin's contracting

>   language could also allow implementation of drivechains.

Reviewer: ```suggestion
  language could also allow the implementation of drivechains.
```

  nodes for each channel, was [announced][mutiny wallet].

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

<!-- FIXME:harding to update Tuesday -->

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Core Lightning #5581][] adds a new event notification topic
  "block_added". Subscribing plugins get notified each time a new block
  is received from bitcoind.

- [Eclair #2418][] and [#2408][eclair #2418] add support for receiving
  payments sent with [blinded routes][topic rv routing].  A spender
  creating a blinded payment isn't provided with the identity of the

>   node receiving the payment, which may improve privacy (especially when
>   used with [unannounced channels][topic unannounced channels]).

Reviewer: ```suggestion
  node receiving the payment, which may improve privacy, especially when
  used with [unannounced channels][topic unannounced channels].
```

    several drawbacks compared to BIP300 but perhaps provides similar
    enough behavior that APO can be considered as enabling drivechains,
    which some individuals may consider a benefit and others may consider
    a problem.

## Changes to services and client software

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

<!-- FIXME:bitschmidty -->

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

<!-- FIXME:harding to update Tuesday -->

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Core Lightning #5581][] Add plugin notification topic "block_processed". FIXME:Xekyo

> - [Eclair #2418][] and [#2408][eclair #2418] add support for receiving

Reviewer 2: I assume that both of the PRs were meant to be linked to? 

```suggestion
- [Eclair #2418][] and [#2408][eclair #2408] add support for receiving
```

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

- **Mempool Project launches Lightning Network explorer:**
  Mempool's open source [Lightning dashboard][mempool lightning] shows aggregate
  network statistics as well as individual node liquidity and connectivity data.

- **Federation software Fedimint adds Lightning:**
  In a recent [blog post][blockstream blog fedimint], Blockstream outlines
  updates to the [Fedimint][] federated Chaumian e-cash project, including
  Lightning Network support.

- **Bitpay wallet improves RBF support:**
  Bitpay [improves][bitpay 12051] its [existing][bitpay 11935] support for

>   sending [RBF][topic rbf] transactions.

Reviewer 2: The commit is not particularly well-explained, but I have the impression that this is a bug fix. It seems to me that if the RBF function was called on a multi-recipient transaction it would only send to a single recipient in the replacement transaction, and now they support bumping multi-recipien transactions. I'm not sure if that's worth mentioning, but if it is, we might want to give a slightly more specific description?

Reviewer 3: Yeah, that is my take as well. I mostly used the recent fix commit as a way to point out the RBF support that was added previously, but apparently I missed it back then. I pushed a small change to add to this description.

@harding Im cool with you dropping this entry, your call!

Author: I'd slightly prefer to keep it in its current edited form.  I think it's ok in this section to give a hat tip to a provider implementing a useful feature without going into details about a specific bug in their implementation.

- **Bitpay wallet improves RBF support:**
  Bitpay [improves][bitpay 12051] its [existing][bitpay 11935] support for

>   sending [RBF][topic rbf] transactions by better handling bumping of multiple
>   output transactions.

Reviewer 2: Sounds much better.
Perhaps:
```suggestion
  sending [RBF][topic rbf] transactions by better handling bumping of 
  transactions with multiple recipients.
```
